libcaja-private/caja-file-operations.c

```diff
@@ -3074,21 +3074,17 @@ report_copy_progress (CopyMoveJob *copy_job,
 	total_size = MAX (source_info->num_bytes, transfer_info->num_bytes);
 
 	elapsed = g_timer_elapsed (job->time, NULL);
-	transfer_rate = 0;
-	if (elapsed > 0) {
-		transfer_rate = transfer_info->num_bytes / elapsed;
-	}
-
-	if (elapsed < SECONDS_NEEDED_FOR_RELIABLE_TRANSFER_RATE &&
-	    transfer_rate > 0) {
+	if (elapsed < SECONDS_NEEDED_FOR_RELIABLE_TRANSFER_RATE) {
 		char *s;
 		/* Translators: %S will expand to a size like "2 bytes" or "3 MB", so something like "4 kb of 4 MB" */
 		s = f (_("%S of %S"), transfer_info->num_bytes, total_size);
 		caja_progress_info_take_details (job->progress, s);
 	} else {
+		double transfer_rate;
 		int remaining_time;
 		char *s;
 
+		transfer_rate = transfer_info->num_bytes / elapsed;
 		remaining_time = (total_size - transfer_info->num_bytes) / transfer_rate;
```

Comment:
This is dangerous, yet nothing really new to this code. if either `elapsed` or `transfer_rate` is 0, it'll crash.
Admittedly, `elapsed` will not be 0 *if `SECONDS_NEEDED_FOR_RELIABLE_TRANSFER_RATE` is greater than 0*, which is gonna be the case but is not necessarily a good thing to depend on.

`transfer_rate` not being 0 puzzles me a tad more, because it means we have to have transferred *something* before this can be safely called.

What about simply changing the condition from `&& transfer_rate > 0` to `|| transfer_rate <= 0`? This fixes the zero-division, and might even have been the intention.

Reply:
Somehow I didn't see that. I didn't get a crash of all of caja but DID get a general failure of the progress indicator to function. In general, any input that could be zero cannot be divided by without checking for and avoiding the zero case.

Reply:
Indeed `SECONDS_NEEDED_FOR_RELIABLE_TRANSFER_RATE` could be changed to zero at a later time, so this shouldn't be assumed to be non-zero. Being a fan of minimally invasive fixes, I reverted the original fix (feel free to discard the original fix and the revert when merging) and instead just changed the condition, as proposed.

As the possible division-by-0 also affects the delete case, I added a fix for this in a separate commit.

Reply:
Ok, this definetly needs to be squashed during merge.
To much revert and fixes of fixes commits.
@basicmaster 
with using git rebase -i you can edit/delete, sort, etc. very easy
Still wondering why coder don't know this strong git command....

Reply:
Could just be new to it: I had to learn these things one at a time myself.

Reply:
I'm aware of interactive rebases but didn't want to do a forced push here, as I didn't know whether/how GitHub handles this in a MR. Thus I pushed the revert commit and the improved solution commits, and asked you to discard them later.
As GitHub apparently copes with a force push, I just discarded the initial solution (and its revert) and rebased the branch on current `master`.

Reply:
I was wondering if there is anything else needed to resolve (and in the end to merge) this. Have you been able to check the rebased solution in the meantime?

Reply:
@cwendling Can this conversation be resolved?